CMakeLists.txt

```diff
@@ -183,6 +184,18 @@ include(util)
 # apply those to the project.
 
 
+##################################################
+# set project version
+if(USED_GIT_VERSION)
+	# VERSION_FULL_STRING is the full git describe
+	set(VERSION_FULL_STRING "${PROJECT_VERSION}")
+	# PROJECT_VERSION is MAJOR.MINOR.PATCH.TWEAK with Commit-Count as Tweak
+	STRING(REGEX REPLACE "v(([0-9]+.|[0-9]+)+)-([0-9]+)-g([a-f0-9]+)"
+			"\\1.\\3" PROJECT_VERSION "${VERSION_FULL_STRING}")
+endif()
```

Comment:
if we used the git describe command we extract `major.minor.patch.tweak` from the output and write it into PROJECT_VERSION. Tweak is the commit count. The git describe output stays in VERSION_FULL_STRING for different usecases (c++ / python version set).